trino-aws-proxy/src/main/java/io/trino/aws/proxy/server/security/opa/OpaS3SecurityFacadeProvider.java

```diff
@@ -65,7 +66,12 @@ public S3SecurityFacade securityFacadeForRequest(ParsedS3Request request, Option
 
     private SecurityResponse facade(ParsedS3Request parsedS3Request, Optional<String> lowercaseAction, Optional<Identity> identity)
     {
-        OpaRequest opaRequest = opaS3SecurityMapper.toRequest(parsedS3Request, lowercaseAction, opaServerUri, identity);
+        OpaRequestAdapter opaRequestAdapter = opaS3SecurityMapper.toRequest(parsedS3Request, lowercaseAction, opaServerUri, identity);
+
+        if (opaRequestAdapter instanceof OpaRequestAdapter.SecurityResponseWrapper securityResponseWrapper) {
+            return securityResponseWrapper.securityResponse();
+        }
```

Comment:
We can extract the field directly

```suggestion
        if (opaRequestAdapter instanceof OpaRequestAdapter.SecurityResponseWrapper(SecurityResponse response)) {
            return response;
        }
```